devito/finite_differences/coefficients.py

```diff
@@ -0,0 +1,89 @@
+import sympy
+
+from cached_property import cached_property
+
+from devito.tools import filter_ordered, flatten
+
+__all__ = ['Coefficients']
+
+
+class Coefficients(object):
+    """
+    Devito class for users to define custom finite difference weights.
+    """
+
+    def __init__(self, *args, **kwargs):
+
+        # Need to check args are valid here.    
+
+        Coefficients.data = args
```

Comment:
I don't like this, I have no idea what's going on

Reply:
Yeah, it's not finished so there's no reason you should.